cs/azure/Trainer/Util/Util.cs

```diff
@@ -0,0 +1,159 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="MyArrayPool.cs">
+//   Copyright (c) by respective owners including Yahoo!, Microsoft, and
+//   individual contributors. All rights reserved.  Released under a BSD
+//   license as described in the file LICENSE.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+using Newtonsoft.Json;
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace VowpalWabbit.Azure.Trainer
+{
+    public class MyArrayPool : IArrayPool<char>
+    {
+        private readonly ArrayPool<char> pool;
+
+        public MyArrayPool()
+        {
+            this.pool = ArrayPool<char>.Create();
+        }
+
+        public char[] Rent(int minimumLength)
+        {
+            return this.pool.Rent(minimumLength);
+        }
+
+        public void Return(char[] array)
+        {
+            this.pool.Return(array);
+        }
+    }
+
+   
+    public static class Util
+    {
+        public static unsafe bool ReadDoubleString(char[] chars, int offset, int length, out double acc)
```

Comment:
is this being tested? also I'm curious what kind of perf gain is obtained here vs the built-in Convert.ToDouble or parse ..